File: packages/Server/Application/Content/FeedItem.cs
namespace Schulcast.Application.Content;

public enum FeedItemType { Article, Podcast, Video }

public class FeedItem
{
	[System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "IDE1006", Justification = "This is an indicator for the client to be able to identify the type of the model")]
	public string __typeName__ => GetType().Name;

	public DateTime Date { get; set; }
	public string Title { get; set; } = null!;
	public FeedItemType Type { get; set; }
	public string ImageUrl { get; set; } = null!;
	public string Link { get; set; } = null!;
	public string[] Tags { get; set; } = Array.Empty<string>();
}

File: packages/Client/sdk/models/FeedItem.ts
import { Entity, model } from 'sdk'

export const enum FeedItemType {
	Article,
	Podcast,
	Video,
}

@model('FeedItem')
export class FeedItem extends Entity {
	type = FeedItemType.Article
	date?: MoDate
	title?: string
	imageUrl?: string
	link?: string
	tags = new Array<string>()
}